src/Compiler/pars.fsy

```diff
@@ -4462,6 +4462,30 @@ declExpr:
        let trivia: SynExprLetOrUseBangTrivia = { LetOrUseBangKeyword = rhs parseState 1 ; EqualsRange = Some mEquals }
        SynExpr.LetOrUseBang(spBind, ($1 = "use"), true, $2, $4, $7, $8, m, trivia) }
 
+  | BINDER atomicPatternLongIdent COLON typeWithTypeConstraints EQUALS typedSequentialExprBlock IN opt_OBLOCKSEP moreBinders typedSequentialExprBlock %prec expr_let
```

Comment:
This new rule allows type annotations without requiring parentheses in let!/use! bindings

```fsharp
let! x: int = ...
use! x: int = ...
```
It's worth noting that this is consistent with how patterns are handled in regular let bindings as well - e.g. when you want to annotate individual elements of a tuple, you need parentheses

```fsharp
let (a: int, b: string) = (1, "hello")  // Valid
let a: int, b: string = (1, "hello")  // Not valid

let! (a: int , b: int) = async { return 1, 3 } // Valid
let! a: int , b: int = async { return 1, 3 }  // Invalid
```